pkg/webhook/webhook_roles.go

```diff
@@ -0,0 +1,104 @@
+//
+// Copyright (c) 2019-2020 Red Hat, Inc.
+// This program and the accompanying materials are made
+// available under the terms of the Eclipse Public License 2.0
+// which is available at https://www.eclipse.org/legal/epl-2.0/
+//
+// SPDX-License-Identifier: EPL-2.0
+//
+// Contributors:
+//   Red Hat, Inc. - initial API and implementation
+//
+
+package webhook
+
+import (
+	"context"
+
+	"github.com/devfile/devworkspace-operator/webhook/server"
+	v1 "k8s.io/api/rbac/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	crclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func CreateWebhookRole(client crclient.Client,
+	ctx context.Context) error {
+
+	clusterRole, err := getSpecRoleBinding()
+	if err != nil {
+		return err
+	}
+
+	if err := client.Create(ctx, clusterRole); err != nil {
+		if !apierrors.IsAlreadyExists(err) {
+			return err
+		}
+		existingCfg, err := getClusterRole(ctx, client)
+		if err != nil {
+			return err
+		}
+		clusterRole.ResourceVersion = existingCfg.ResourceVersion
+		err = client.Update(ctx, clusterRole)
+		if err != nil {
+			return err
+		}
+		log.Info("Updated webhook server cluster role")
+	} else {
+		log.Info("Created webhook server cluster role")
+	}
+
+	return nil
+}
+
+func getSpecRoleBinding() (*v1.ClusterRole, error) {
+	clusterRole := &v1.ClusterRole{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:   server.WebhookServerSAName,
+			Labels: server.WebhookServerAppLabels(),
+		},
+		Rules: []v1.PolicyRule{
+			{
+				APIGroups: []string{
+					"admissionregistration.k8s.io",
+				},
+				Resources: []string{
+					"mutatingwebhookconfigurations",
+					"validatingwebhookconfigurations",
+				},
+				Verbs: []string{
+					"*",
```

Comment:
Do we need `delete/deleteCollection`? I worry that taking unnecessary permissions could make people hesitate to install the operator. We'd also need to update `role.yaml` with this change.

Reply:
Better request what we only need

Reply:
Limited to `get, list, create, patch, update`.